### ISSUE TITLE
errors to build arduino firmware 2.0.x

### ISSUE THREAD

**Issue by fgmGitHub** [platformio/platformio-core#1870]
Errors found with Arduino Firmware build 2.0.x (latest version (today))

------------------------------------------------------------------

### Configuration

**Windows 10**:

**PlatformIO Version** (`18.1`):

### Description of problem

Processing megaatmega2560 (platform: atmelavr; board: megaatmega2560; framework: arduino)
----------------------------------------------------------------------------------------------------------------------------------------------
Error: Traceback (most recent call last):
  File "c:\users\fernando\.platformio\penv\lib\site-packages\platformio\__main__.py", line 120, in main
    cli(None, None, None)
  File "c:\users\fernando\.platformio\penv\lib\site-packages\click\core.py", line 700, in __call__
    return self.main(*args, **kwargs)
  File "c:\users\fernando\.platformio\penv\lib\site-packages\click\core.py", line 680, in main
    rv = self.invoke(ctx)
  File "c:\users\fernando\.platformio\penv\lib\site-packages\click\core.py", line 1027, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "c:\users\fernando\.platformio\penv\lib\site-packages\click\core.py", line 873, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "c:\users\fernando\.platformio\penv\lib\site-packages\click\core.py", line 508, in invoke
    return callback(*args, **kwargs)
  File "c:\users\fernando\.platformio\penv\lib\site-packages\click\decorators.py", line 16, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "c:\users\fernando\.platformio\penv\lib\site-packages\platformio\commands\run.py", line 107, in cli
    result = (envname, ep.process())
  File "c:\users\fernando\.platformio\penv\lib\site-packages\platformio\commands\run.py", line 205, in process
    result = self._run()
  File "c:\users\fernando\.platformio\penv\lib\site-packages\platformio\commands\run.py", line 300, in _run
    self.verbose)
  File "c:\users\fernando\.platformio\penv\lib\site-packages\platformio\commands\run.py", line 323, in _autoinstall_libdeps
    ctx.invoke(cmd_lib_install, libraries=[lib], silent=not verbose)
  File "c:\users\fernando\.platformio\penv\lib\site-packages\click\core.py", line 508, in invoke
    return callback(*args, **kwargs)
  File "c:\users\fernando\.platformio\penv\lib\site-packages\click\decorators.py", line 26, in new_func
    return f(get_current_context().obj, *args, **kwargs)
  File "c:\users\fernando\.platformio\penv\lib\site-packages\platformio\commands\lib.py", line 103, in lib_install
    library, silent=silent, interactive=interactive, force=force)
  File "c:\users\fernando\.platformio\penv\lib\site-packages\platformio\managers\lib.py", line 352, in install
    force=force)
  File "c:\users\fernando\.platformio\penv\lib\site-packages\platformio\managers\package.py", line 685, in install
    with LockFile(self.package_dir):
  File "c:\users\fernando\.platformio\penv\lib\site-packages\platformio\lockfile.py", line 102, in __enter__
    self.acquire()
  File "c:\users\fernando\.platformio\penv\lib\site-packages\platformio\lockfile.py", line 86, in acquire
    return self._lock()
  File "c:\users\fernando\.platformio\penv\lib\site-packages\platformio\lockfile.py", line 61, in _lock
    self._fp = open(self._lock_path, "w")
IOError: [Errno 13] Permission denied: u'y:\\Software\\3D_Printer\\Arduino\\Scalar\\Marlin-bugfix-2.0.06.10.2018\\Marlin-bugfix-2.0.x\\.piolibdeps.lock'

============================================================

An unexpected error occurred. Further steps:

* Verify that you have the latest version of PlatformIO using
  `pip install -U platformio` command

* Try to find answer in FAQ Troubleshooting section
  https://docs.platformio.org/page/faq.html

* Report this problem to the developers
  https://github.com/platformio/platformio-core/issues

============================================================

O processo terminal encerrado com código de saída: 1

Terminal será reutilizado pelas tarefas, pressione qualquer tecla para fechar.

### If problems with PlatformIO Build System:

**The content of `platformio.ini`:**
```
#
# Marlin Firmware
# PlatformIO Configuration File
#
# For detailed documentation with EXAMPLES:
#
# http://docs.platformio.org/en/latest/projectconf.html
#

# Automatic targets - enable auto-uploading
# targets = upload

#
# By default platformio build will abort after 5 errors.
# Remove '-fmax-errors=5' from build_flags below to see all.
#

[platformio]
src_dir     = Marlin
build_dir   = .pioenvs
lib_dir     = .piolib
libdeps_dir = .piolibdeps
boards_dir  = buildroot/share/PlatformIO/boards
env_default = megaatmega2560

[common]
default_src_filter = +<src/*> -<src/config> -<src/HAL> +<src/HAL/shared>
build_flags = -fmax-errors=5
  -g
  -ggdb
lib_deps =
  https://github.com/MarlinFirmware/U8glib-HAL/archive/dev.zip
  LiquidCrystal@1.3.4
  https://github.com/teemuatlut/TMCStepper.git
  Adafruit NeoPixel@1.1.3
  https://github.com/lincomatic/LiquidTWI2/archive/30aa480.zip
  https://github.com/ameyer/Arduino-L6470/archive/master.zip
  https://github.com/trinamic/TMC26XStepper/archive/c1921b4.zip

#################################
#                               #
#   Unique Core Architectures   #
#                               #
#  Add a new "env" below if no  #
# entry has values suitable to  #
#   build for a given board.    #
#                               #
#################################

#
# ATmega2560
#
[env:megaatmega2560]
platform          = atmelavr
framework         = arduino
board             = megaatmega2560
build_flags       = ${common.build_flags}
board_build.f_cpu = 16000000L
lib_deps          = ${common.lib_deps}
src_filter        = ${common.default_src_filter} +<src/HAL/HAL_AVR>
monitor_speed     = 250000

#
# ATmega1280
#
[env:megaatmega1280]
platform          = atmelavr
framework         = arduino
board             = megaatmega1280
build_flags       = ${common.build_flags}
board_build.f_cpu = 16000000L
lib_deps          = ${common.lib_deps}
src_filter        = ${common.default_src_filter} +<src/HAL/HAL_AVR>
monitor_speed     = 250000

#
# AT90USB1286 boards using CDC bootloader
# - BRAINWAVE
# - BRAINWAVE_PRO
# - SAV_MKI
# - TEENSYLU
#
[env:at90usb1286_cdc]
platform      = teensy
framework     = arduino
board         = at90usb1286
build_flags   = ${common.build_flags}
lib_deps      = ${common.lib_deps}
lib_ldf_mode  = deep+
src_filter    = ${common.default_src_filter} +<src/HAL/HAL_AVR>
extra_scripts = pre:buildroot/share/atom/create_custom_upload_command_CDC.py
monitor_speed = 250000

#
# AT90USB1286 boards using DFU bootloader
# - PrintrBoard
# - PrintrBoard Rev.F
# - ? 5DPRINT ?
#
[env:at90usb1286_dfu]
platform      = teensy
framework     = arduino
board         = at90usb1286
build_flags   = ${common.build_flags}
lib_deps      = ${common.lib_deps}
lib_ldf_mode  = deep+
src_filter    = ${common.default_src_filter} +<src/HAL/HAL_AVR>
extra_scripts = pre:buildroot/share/atom/create_custom_upload_command_DFU.py
monitor_speed = 250000

#
# Due (Atmel SAM3X8E ARM Cortex-M3)
#
#  - RAMPS4DUE
#  - RADDS
#
[env:DUE]
platform      = atmelsam
framework     = arduino
board         = due
build_flags   = ${common.build_flags}
lib_deps      = ${common.lib_deps}
lib_ignore    = c1921b4
src_filter    = ${common.default_src_filter} +<src/HAL/HAL_DUE>
monitor_speed = 250000
[env:DUE_USB]
platform      = atmelsam
framework     = arduino
board         = dueUSB
build_flags   = ${common.build_flags}
lib_deps      = ${common.lib_deps}
lib_ignore    = c1921b4
src_filter    = ${common.default_src_filter} +<src/HAL/HAL_DUE>
monitor_speed = 250000
[env:DUE_debug]
# Used when WATCHDOG_RESET_MANUAL is enabled
platform      = atmelsam
framework     = arduino
board         = due
build_flags   = ${common.build_flags}
  -funwind-tables
  -mpoke-function-name
lib_deps      = ${common.lib_deps}
lib_ignore    = c1921b4
src_filter    = ${common.default_src_filter} +<src/HAL/HAL_DUE>
monitor_speed = 250000

#
# NXP LPC1768 ARM Cortex-M3
#
[env:LPC1768]
platform          = nxplpc@<3.4.0
board             = lpc1768
board_build.f_cpu = 100000000L
# Override default maximum RAM. LPC1768/9 do have 64k, but in 3 blocks (32K, 16K, 16K).
# The first 32k block is used by default, while the others must be specifically targeted.
board_upload.maximum_ram_size = 32768
build_flags       = !python Marlin/src/HAL/HAL_LPC1768/lpc1768_flag_script.py ${common.build_flags}
build_unflags     = -Wall
lib_ldf_mode      = off
lib_extra_dirs    = frameworks
lib_deps          = CMSIS-LPC1768
  https://github.com/MarlinFirmware/U8glib-HAL/archive/dev.zip
  https://github.com/teemuatlut/TMCStepper.git
extra_scripts     = Marlin/src/HAL/HAL_LPC1768/debug_extra_script.py, Marlin/src/HAL/HAL_LPC1768/lpc1768_flag_script.py, Marlin/src/HAL/HAL_LPC1768/upload_extra_script.py
src_filter        = ${common.default_src_filter} +<src/HAL/HAL_LPC1768>
monitor_speed     = 250000
debug_tool        = custom
debug_server      =
  C:\Program Files (x86)\SEGGER\JLink_V618d\JLinkGDBServerCL.exe
  -select USB
  -port 2331
  -device LPC1768
  -if JTAG
  -speed auto
  -noir

#
# Melzi and clones (ATmega1284p)
#
[env:melzi]
platform      = atmelavr
framework     = arduino
board         = sanguino_atmega1284p
build_flags   = ${common.build_flags}
upload_speed  = 57600
lib_deps      = ${common.lib_deps}
src_filter    = ${common.default_src_filter} +<src/HAL/HAL_AVR>
monitor_speed = 250000

#
# Melzi and clones (Optiboot bootloader)
#
[env:melzi_optiboot]
platform      = atmelavr
framework     = arduino
board         = sanguino_atmega1284p
build_flags   = ${common.build_flags}
upload_speed  = 115200
lib_deps      = ${common.lib_deps}
src_filter    = ${common.default_src_filter} +<src/HAL/HAL_AVR>
monitor_speed = 250000

#
# RAMBo
#
[env:rambo]
platform          = atmelavr
framework         = arduino
board             = reprap_rambo
build_flags       = ${common.build_flags}
board_build.f_cpu = 16000000L
lib_deps          = ${common.lib_deps}
src_filter        = ${common.default_src_filter} +<src/HAL/HAL_AVR>
monitor_speed     = 250000

#
# Sanguinololu (ATmega644p)
#
[env:sanguino_atmega644p]
platform      = atmelavr
framework     = arduino
board         = sanguino_atmega644p
build_flags   = ${common.build_flags}
lib_deps      = ${common.lib_deps}
src_filter    = ${common.default_src_filter} +<src/HAL/HAL_AVR>
monitor_speed = 250000

#
# Sanguinololu (ATmega1284p)
#
[env:sanguino_atmega1284p]
platform      = atmelavr
framework     = arduino
board         = sanguino_atmega1284p
build_flags   = ${common.build_flags}
lib_deps      = ${common.lib_deps}
src_filter    = ${common.default_src_filter} +<src/HAL/HAL_AVR>
monitor_speed = 250000

#
# STM32F103RE
#
[env:STM32F1]
platform      = ststm32@<4.4.0
framework     = arduino
board         = genericSTM32F103RE
build_flags   = !python Marlin/src/HAL/HAL_STM32F1/stm32f1_flag_script.py
  ${common.build_flags}
lib_deps      = ${common.lib_deps}
lib_ignore    = U8glib-HAL
  c1921b4
  libf3c
  lib066
  Adafruit NeoPixel_ID28
  Adafruit NeoPixel
  libf3e
  TMC26XStepper
lib_ldf_mode  = 1
src_filter    = ${common.default_src_filter} +<src/HAL/HAL_STM32F1>
monitor_speed = 250000

#
# STM32F4
#
[env:STM32F4]
platform      = ststm32
framework     = arduino
board         = disco_f407vg
build_flags   = ${common.build_flags} -DUSE_STM32GENERIC -DSTM32GENERIC -DMENU_USB_SERIAL -DMENU_SERIAL=SerialUSB
lib_deps      = ${common.lib_deps}
lib_ignore    = Adafruit NeoPixel, c1921b4, TMCStepper
src_filter    = ${common.default_src_filter} +<src/HAL/HAL_STM32F4>
monitor_speed = 250000

#
# Teensy 3.5 / 3.6 (ARM Cortex-M4)
#
[env:teensy35]
platform      = teensy
framework     = arduino
board         = teensy35
build_flags   = ${common.build_flags}
lib_deps      = ${common.lib_deps}
lib_ignore    = Adafruit NeoPixel
src_filter    = ${common.default_src_filter} +<src/HAL/HAL_TEENSY35_36>
monitor_speed = 250000

[env:malyanm200]
platform    = ststm32
framework   = arduino
board       = malyanM200
build_flags = !python Marlin/src/HAL/HAL_STM32F1/stm32f1_flag_script.py -DMCU_STM32F103CB -D __STM32F1__=1 -std=c++1y -D MOTHERBOARD="BOARD_MALYAN_M200" -DSERIAL_USB -ffunction-sections -fdata-sections -Wl,--gc-sections
src_filter  = ${common.default_src_filter} +<src/HAL/HAL_STM32F1>
#-<frameworks>
lib_ignore  =
  U8glib
  LiquidCrystal_I2C
  LiquidCrystal
  NewliquidCrystal
  LiquidTWI2
  Adafruit NeoPixel
  TMCStepper
  Servo(STM32F1)
  TMC26XStepper
  U8glib-HAL
  c1921b4

#
# Espressif ESP32
#
[env:esp32]
platform    = https://github.com/platformio/platform-espressif32.git#feature/stage
board       = esp32dev
framework   = arduino
upload_port = COM3
lib_ignore  =
  LiquidCrystal_I2C
  LiquidCrystal
  NewliquidCrystal
  LiquidTWI2
  TMC26XStepper
  c1921b4
src_filter  = ${common.default_src_filter} +<src/HAL/HAL_ESP32>

```

**Comment by ivankravets** [comment]
> IOError: [Errno 13] Permission denied: u'y:\Software\3D_Printer\Arduino\Scalar\Marlin-bugfix-2.0.06.10.2018\Marlin-bugfix-2.0.x\.piolibdeps.lock'

Please remove `y:\Software\3D_Printer\Arduino\Scalar\Marlin-bugfix-2.0.06.10.2018\Marlin-bugfix-2.0.x\.piolibdeps.lock` and try again.

**Comment by fgmGitHub** [comment]
Thanks



Fernando G. Monteiro |
Enfermeiro Chefe do Hospital de Dia - Adultos
Vogal de Direção da EPOP
[1502106312236_image003.png]

 <http://www.sns.gov.pt/>  <http://www.sns.gov.pt/>
________________________________
De: Ivan Kravets <notifications@github.com>
Enviado: 10 de outubro de 2018 11:32:55
Para: platformio/platformio-core
Cc: Fernando G. Monteiro; Author
Assunto: Re: [platformio/platformio-core] errors to build arduino firmware 2.0.x (#1870)


Closed #1870<https://github.com/platformio/platformio-core/issues/1870>.
